src/network/socket.cpp

```diff
@@ -97,9 +98,17 @@ bool UDPSocket::init(bool ipv6, bool noExceptions)
 		return false;
 	}
 
-	// Use IPv6 if specified
-	m_addr_family = ipv6 ? AF_INET6 : AF_INET;
-	m_handle = socket(m_addr_family, SOCK_DGRAM, IPPROTO_UDP);
+	int sd_sockets = sd_listen_fds(0);
+	if (sd_sockets > 1)
+		dstream << "Too many file descriptors received." << std::endl;
+	else if (sd_sockets == 1) {
+		m_addr_family = AF_INET6;
+		m_handle = SD_LISTEN_FDS_START + 0;
```

Comment:
When the socket is inherited, you probably either want to skip the close() in the destructor of UDPSocket, or call dup(SD_LISTEN_FDS_START) here to make a duplicate of the fd that you can safely close.